modules/flashprog

```diff
@@ -2,13 +2,11 @@ modules-$(CONFIG_FLASHPROG) += flashprog
 
 flashprog_depends := pciutils $(musl_dep)
 
-flashprog_change := 72991
-flashprog_revision := 5
-flashprog_version := $(flashprog_change)_$(flashprog_revision)
+flashprog_version := 9dc6d843b0678001c9baf0e8602ecb25b16329d2
 flashprog_dir := flashprog-$(flashprog_version)
 flashprog_tar := $(flashprog_dir).tar.gz
-flashprog_url := https://review.sourcearcade.org/changes/flashprog~$(flashprog_change)/revisions/$(flashprog_revision)/archive?format=tgz
-flashprog_hash := 8e8e6d901d3bc37b80db1f10dd21b9490709bdb54a0ca06892f8922a284f2ed9
+flashprog_url := https://github.com/SourceArcade/flashprog/archive/$(flashprog_version).tar.gz
+flashprog_hash := fa4ddf3b60314994a37e4599122ae4c7f42135c13c782e580bc580d715cfa2fc
```

Comment:
It's sitting ducks for review. I don't expect bigger changes.
@SergiiDmytruk know any users of the block protection that could give some feedback on the CLI?

Reply:
Haha, yeah, it's not just a give-me-security switch. Modern chips don't even have a /WP pin (because they prefer quad-i/o). Biggest bummer though is the lacking support to read/write all registers on Intel platforms (I have one more idea to try but little hope), otherwise one could use the protect-until-powercycle feature.

@ln2max, @jtf7 looking for user test and feedback on the WP CLI available through this PR, or [this branch](https://github.com/SourceArcade/flashprog/tree/wp_cli). It's not too different from flashrom's, e.g. `wp --status` instead of `--wp-status`. And it supports only one command at a time to be sure about the execution order.

Reply:
No, there's another feature of the flash chip itself. Most common for SPI flashes is that you can configure which blocks to protect and then there are four states:

1. the configuration is unprotected,
2. the configuration is locked by the /WP pin,
3. the configuration is locked until a power cycle, or
4. the configuration is permanently locked.

For 2. we need the pin and support or modification of the mainboard. 3. and 4. often require to write a bit in a secondary register. Problem is so far nobody knows how to write that register with Intel's `hwseq`, which is the only option since Skylake. I don't think Intel implemented that. They just want you to use their protection mechanisms like PR0; those are, however, very inconvenient when Intel FSP is involved.

Something like that hardenedvault patch wouldn't work on (at least some) newer systems because FSP would lock PR0 too early, unconditionally. Security with Intel is never easy.

Reply:
> the configuration is locked until a power cycle

I thought chips supporting this feature are fairly uncommon.  I remember seeing notes in a bunch of datasheets to the effect of "this feature is available per request" (maybe all from a single vendor).

Reply:
Hmmm, my recollection may be biased by our own chip selection (we used this feature on an AMD based system almost 15 years ago). I thought it's very common for Winbond; and GD and XMC for instance seem to copy everything from them judging from datasheets. Macronix seems to be an exception. This "available per request" does ring a bell, but I usually see this regarding 4., the permanent locking.

Reply:
@i-c-o-n you might want to look at https://github.com/linuxboot/heads/pull/1818 which thanks to @miczyg1 patchset over dasharo fork, brings PR0 to Skylake+

Will need to update https://github.com/linuxboot/heads/pull/1769/files#diff-546b89876634c383a0dd21ef655ef6d3c5c62c69e33a2f6910dea84ac1d49eea to reflect this fact, since merging this PR will happen soon. Just need to make sure newer boards addition (optiplex only if I recall well) also need to be switched to flashprog and work as expected (should be).

@i-c-o-n : should I point to newer flashprog commit in this PR, or do so in another PR?